### PR TITLE
Makefile: Add "propagate-version" target

### PR DESCRIPTION
**Pull request by ldoktor**
This commit modifies the optional_plugins to depend on VERSION files
located in their main directories and adds "propagate-version" target to
Makefile to update the VERSION files for optional as well as "linkable"
plugins, if VERSION file was present.

Signed-off-by: Lukáš Doktor <ldoktor@redhat.com>

### CODE REVIEW

**Comment by ldoktor** [comment]
[Add script to update VERSION for all parts of Avocado](https://trello.com/c/Lpz4XcZX/988-add-script-to-update-version-for-all-parts-of-avocado)

File: docs/source/MaintenanceGuide.rst [hunk 1]
```diff
@@ -11,119 +11,84 @@ details you need to pay attention to.
...
-For the core Avocado versioning, two files need to receive a version update:
+For the Avocado versioning, two files need to receive a manual version update:
 
  * ``VERSION``
  * ``python-avocado.spec``
 
-So, go through the Avocado code base and update the version number. At
-the time of this writing, the diff looks like this:
+followed by ``make propagate-version`` to propagate this change to all
+optional and "linkabe" plugins sharing the parent dir (eg. ``avocado-vt``).
```
Comment (clebergnu) [inline docs/source/MaintenanceGuide.rst]: Typo: "linkable"

**Comment by ldoktor** [comment]
Are you sure? It works well for RPM-installed-version as well as for `make develop`. I checked it by using:

```python
import pkg_resources
pkg_resources.require("avocado-framework-plugin-runner-remote")[0].version
```
and it worked well (changing the version propagated)

**Comment by clebergnu** [comment]
Yep, as I mentioned before, the optional plugin **tarballs** are created like that.  They are used on pip-like installs, but not on source code or RPM based installs.

**Comment by ldoktor** [comment]
I see, they are missing in `make pypi`. I addressed this issue in the next PR, hopefully everything will be fine then.

**Pull request closed without merging**